### PR TITLE
Add parsing support for JSON Search Functions

### PR DESCRIPTION
**Pull request by K-Kumar-01**
## Description

Add the parsing support for JSON search functions. More details [here](https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html)

Need discussion on `MEMBER OF` .

## Related Issue(s)

#8604 
#4099 

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -6190,6 +6256,12 @@ non_reserved_keyword:
 | JSON_ARRAY
 | JSON_OBJECT
 | JSON_QUOTE
+| JSON_CONTAINS
+| JSON_CONTAINS_PATH
+| JSON_EXTRACT
+| JSON_KEYS
+| JSON_SEARCH
+| JSON_VALUE
```
Comment (GuptaManan100) [inline go/vt/sqlparser/sql.y:6264]: You missed JSON_OVERLAPS

File: go/vt/sqlparser/sql.y [hunk 2]
```diff
@@ -6190,6 +6256,12 @@ non_reserved_keyword:
 | JSON_ARRAY
 | JSON_OBJECT
 | JSON_QUOTE
+| JSON_CONTAINS
```
Comment (GuptaManan100) [inline go/vt/sqlparser/sql.y:6259]: This should be alphabetically sorted.

File: go/vt/sqlparser/keywords.go [hunk 3]
```diff
@@ -318,6 +318,13 @@ var keywords = []keyword{
...
+	{"json_contains", JSON_CONTAINS},
+	{"json_contains_path", JSON_CONTAINS_PATH},
+	{"json_extract", JSON_EXTRACT},
+	{"json_keys", JSON_KEYS},
+	{"json_overlaps", JSON_OVERLAPS},
+	{"json_search", JSON_SEARCH},
+	{"json_value", JSON_VALUE},
 	{"json_pretty", JSON_PRETTY},
 	{"json_storage_size", JSON_STORAGE_SIZE},
 	{"json_storage_free", JSON_STORAGE_FREE},
```
Comment (GuptaManan100) [inline go/vt/sqlparser/keywords.go:330]: This should also be alphabetically sorted

**Review by GuptaManan100 — COMMENTED** [review]
We should also add tests for the shorthands `->` and `->>`

File: go/vt/sqlparser/keywords.go (hunk 3 above)
Comment (K-Kumar-01) [reply go/vt/sqlparser/keywords.go:330]: > We should also add tests for the shorthands `->` and `->>`

There were some tests already present so I didn't add them.

Comment (derekperkins) [reply go/vt/sqlparser/keywords.go]: This might not be the place for the tests, but there was a regression in v13 #9951

Comment (GuptaManan100) [reply go/vt/sqlparser/keywords.go]: @derekperkins I plan to look at the issue before merging this PR today. I already tried running the query through the parser mentioned in #9951 and it passed. I think the issue might be in the planner, but I'll know more today and update on the issue

**Review by GuptaManan100 — APPROVED** [review]

**Pull request merged**